domain/model/modelmigration/export.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package modelmigration
+
+import (
+	"context"
+
+	"github.com/juju/description/v8"
+
+	"github.com/juju/juju/core/logger"
+	"github.com/juju/juju/core/model"
+	coremodel "github.com/juju/juju/core/model"
+	"github.com/juju/juju/core/modelmigration"
+	"github.com/juju/juju/domain/model/service"
+	"github.com/juju/juju/domain/model/state"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/internal/errors"
+)
+
+// RegisterExport registers the export operations with the given coordinator.
+func RegisterExport(coordinator Coordinator, logger logger.Logger) {
+	coordinator.Add(&exportOperation{
+		logger: logger,
+	})
+}
+
+// ExportService provides a subset of the model domain
+// service methods needed for model export.
+type ExportService interface {
+	// GetEnvironProviderVersion retrieves the version of the environment provider
+	// associated with the model.
+	GetEnvironProviderVersion(context.Context) (int, error)
+}
+
+// exportOperation describes a way to execute a migration for
+// exporting model.
+type exportOperation struct {
+	modelmigration.BaseOperation
+
+	serviceGetter func(modelUUID model.UUID) ExportService
+	logger        logger.Logger
+}
+
+// Name returns the name of this operation.
+func (e *exportOperation) Name() string {
+	return "export model"
+}
+
+// Setup the export operation, this will ensure the service is created
+// and ready to be used.
+func (e *exportOperation) Setup(scope modelmigration.Scope) error {
+	e.serviceGetter = func(modelUUID model.UUID) ExportService {
```

Comment:
We just create the service here, no need to have this service getter.
I know you may ask how we get the model UUID here.
I don't think the ModelServie should inject the model UUID at all, it can always fetch the model UUID from DB.
Or you can just create a new service type for migration like the `domain/application`(this service be a very simple one):
https://github.com/juju/juju/blob/bd7f9f4639328f76693d1357e03ff60fbe42d700/domain/application/modelmigration/import.go#L82

Reply:
Hi kelvin, from my understand we usually fetch the model from the DB using the model UUID, can I check what information would we use to fetch the model UUID from the DB?

Reply:
Hey Tom, thanks for the clarification. Just wondering does it mean each model has its own model DB that's why we could directly retrieve its UUID from the db?

Reply:
There's an implementation mistake that needs to be fixed, but not in this PR. The model UUID is used for model CRUD, which needs to be implemented in a separate "model manager" service. Once that is done, we can revisit this and use the same pattern of directly setting the service as is used in the other 11 exporters.

We've not yet got a risk register item - please add one and a add a TODO in the code, eg
```
// TODO(alvin): we need to extract model creation/deletion into a separate service so that this 
// service operates only on the current model and we no longer need to pass in the UUID when
// creating the service.
```

Reply:
Just to add to this we have a task in the core model epic to deal with this item.

Reply:
Sure thanks Ian and tom